prov/verbs/src/fi_verbs.h

```diff
@@ -465,16 +465,15 @@ struct fi_ibv_srq_ep {
 	struct ibv_srq		*srq;
 	struct fi_ibv_domain	*domain;
 
-	/* Due to XRC SRQ semantics, an XRC SRX context may only be bound
-	 * to multiple endpoints when they share the same RX CQ. */
-	struct fi_ibv_cq	*srq_cq;
-
-	/* XRC SRQ is not created until endpoint enable */
-	fastlock_t		prepost_lock;
-	struct slist		prepost_list;
-	uint32_t		max_recv_wr;
-	uint32_t		max_sge;
-	uint32_t		prepost_count;
+	/* For XRC SRQ only */
+	struct {
+		/* XRC SRQ is not created until endpoint enable */
+		fastlock_t		prepost_lock;
```

Comment:
We can use the verbs:util ep lock here.  See util_ep::lock_acquire/lock_release functions, which are set based on the app threading model

Reply:
The pre-existing fi_ibv_srq_ep structure is lightweight and does not include a util_ep object? Since the fi_srx_context is bound to multiple EP, I don't think we can use the util_ep lock in the individual EP either.